src/Compilers/Core/Portable/SourceFileResolver.cs

```diff
@@ -62,9 +62,9 @@ public SourceFileResolver(ImmutableArray<string> searchPaths, string baseDirecto
                     }
 
                     var value = kv.Value;
-                    if (value == null || value.Length == 0)
+                    if (value == null)
                     {
-                        throw new ArgumentException(CodeAnalysisResources.EmptyValueInPathMap, nameof(pathMap));
+                        throw new ArgumentNullException(nameof(PathMap));
```

Comment:
An exception has changed, but none of the unit-tests caught that. Please create a unit-test covering this code path.